index.js

```diff
@@ -289,6 +291,10 @@ function normalizeArguments(url, opts) {
 		opts
 	);
 
+	if (!opts.decompress) {
+		opts.encoding = null;
+	}
```

Comment:
Fixed. Also added test to check `opts.decompress` doesn't alter encoding of uncompressed responses.